x-pack/plugins/security/server/lib/saved_objects_client/secure_saved_objects_client.js

```diff
@@ -69,28 +63,26 @@ export class SecureSavedObjectsClient {
   }
 
   async update(type, id, attributes, options = {}) {
-    await this._performAuthorizationCheck(type, attributes, options);
+    await this._performAuthorizationCheck(type, 'update');
 
     return await this._client.update(type, id, attributes, options);
   }
 
-  async _performAuthorizationCheck(type, action, attributes = {}, options = {}) { // eslint-disable-line no-unused-vars
-    return;
-    // TODO(legrego) use ES Custom Privilege API once implemented.
-    const kibanaAction = `saved-objects/${type}/${action}`;
-
-    const privilegeCheck = await this._callCluster(this._request, 'shield.hasPrivileges', {
-      body: {
-        applications: [{
-          application: this._application,
-          resources: ['default'],
-          privileges: [kibanaAction]
-        }]
-      }
-    });
-
-    if (!privilegeCheck.has_all_requested) {
-      throw Boom.unauthorized(`User ${privilegeCheck.username} is not authorized to ${action} objects of type ${type}`);
+  async _performAuthorizationCheck(typeOrTypes, action) {
+    const types = Array.isArray(typeOrTypes) ? typeOrTypes : [typeOrTypes];
+    const actions = types.map(type => `action:saved-objects/${type}/${action}`);
+
+    let result;
+    try {
+      result = await this._hasPrivileges(actions);
+    } catch(error) {
+      const { reason } = get(error, 'body.error', {});
```

Comment:
If we ever get an error thrown when checking the `hasPrivileges`, I'm considering it a "general error" so it'll surface as a 500. We should've authenticated the user before this point, and we won't get a 403 (as this will just return true/false) and all other exceptions are something we don't want to surface to the user via customized HTTP access codes.

Reply:
We're missing a lot of tests in the RBAC Phase 1 branch, I'm going to be working on adding better test coverage in an alternate PR (since I've already done a bit too much in this one).